backend/activity.go

```diff
@@ -68,6 +68,8 @@ func (p *activityProcessor) ProcessWorkItem(ctx context.Context, wi WorkItem) er
 		}()
 	}
 
+	// set the parent trace context to be the newly created activity span
+	ts.ParentTraceContext = helpers.TraceContextFromSpan(span)
```

Comment:
No need to reset the `ParentTraceContext` here as the newly created context above already has the span information. It can be extracted in the `ExecuteActivity` itself.

Reply:
I think I still need to set the `ParentTraceContext` as I need to populate it here https://github.com/microsoft/durabletask-go/blob/d11688c289f5ab182341534f2312a543bd82615f/backend/executor.go#L151

Reply:
We are actually passing the new context to `ExecuteActivity` right? So we can get `ParentTraceContext` in the `ExecuteActivity` itself.

It possibly might be okay to update in the event as well, my only concern is that since event gets persisted as well, it would be permanently stored as the new `ParentTraceContext` for the task and it might cause issues in case of errors/retries. For now I don't see any issue directly though.

Reply:
We can actually get span from context itself: https://pkg.go.dev/go.opentelemetry.io/otel/trace#SpanFromContext and then get traceContext from span, but i guess it's easier updating the event itself.

My issue was updating the event was that it should't cause any issue in case of errors/retries(like recursively adding a new span under activity span), but it doesn't seem anything like that happening for now, so no issues.